### PR TITLE
Feature/66 Fix design/responsiveness related bugs  

### LINKED ISSUE #66: Responsive layout for novel page

### PR DESCRIPTION
**Pull request by moetazaneta**
Closes #66

### CODE REVIEW

File: src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx [hunk 1]
```diff
@@ -59,6 +60,21 @@ export const CharactersPage: VFC = () => {
...
+  let columnCount = 4;
+  if (canFit4OrMore) {
+    columnCount = 4;
+  } else if (canFitOnly3) {
+    columnCount = 3;
+  } else if (canFitOnly2) {
+    columnCount = 2;
+  } else {
+    columnCount = 1;
+  }
```
Comment (NightOfNights) [inline src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx]: You can avoid `let` and `else if` here if you move out this code part as function

File: src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx [hunk 2]
```diff
@@ -59,6 +60,21 @@ export const CharactersPage: VFC = () => {
     return groupedCharacters;
   };
 
+  const [canFit4OrMore, canFitOnly3, canFitOnly2] = useMediaQuery(
+    ['(min-width: 1564px)', '(min-width: 1280px)', '(min-width: 1024px)'],
+  );
```
Comment (NightOfNights) [inline src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx]: Bad and implicit naming, what if we would want to use media query hook in this component for purposes other than calculating column amount?

**Review by NightOfNights — CHANGES_REQUESTED** [review]

File: src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx (hunk 2 above)
Comment (moetazaneta) [reply src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx]: The sizes are specific for calculating column amount and shouldn't be used anywhere else.

File: src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx [hunk 3]
```diff
@@ -59,6 +60,24 @@ export const CharactersPage: VFC = () => {
...
+      return 4;
+    }
+    if (canFitOnly3) {
+      return 3;
+    }
+    if (canFitOnly2) {
+      return 2;
+    }
+    return 1;
+  };
```
Comment (NightOfNights) [inline src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx]: Please, [don't call hooks inside functions](https://ru.reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level)

Comment (NightOfNights) [reply src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx]: Btw, I have this error if I to try to open [http://localhost:3000/vn/7/characters](http://localhost:3000/vn/7/characters) directly
image

Comment (NightOfNights) [reply src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx]: And also you can compare with profiler or `console.log` how many times component rerenders when `useMediaQuery` hook is inside the function and when it's outside :)

File: src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx [hunk 4]
```diff
@@ -59,6 +60,24 @@ export const CharactersPage: VFC = () => {
     return groupedCharacters;
   };
 
+  /** Calculates character column count. */
+  const getCharactersColumnCount = (): number => {
+    const [canFit4OrMore, canFitOnly3, canFitOnly2] = useMediaQuery(
```
Comment (NightOfNights) [inline src/features/VisualNovel/pages/CharactersPage/CharactersPage.tsx]: I am still thinking that names like `isScreenWidthEqualOrLargerThan1024px` would be more semantically correct and reusable.

**Review by NightOfNights — CHANGES_REQUESTED** [review]

**Pull request merged**